packaging/language/pear.py

```diff
@@ -149,8 +149,11 @@ def install_packages(module, state, packages):
         cmd = "pear %s %s" % (command, package)
         rc, stdout, stderr = module.run_command(cmd, check_rc=False)
 
-        if rc != 0:
-            module.fail_json(msg="failed to install %s" % (package))
+        # verify that the package is actually installed
+        # as pear exit value is unreliable
+        installed, updated = query_package(module, package)
+        if rc != 0 or not installed or (state == 'latest' and not updated):
```

Comment:
The "updated" variable is true if the latest version of the package is installed. So this seems OK to me, if the required package state is "latest" and the package is not in the latest version, throw an error.

(But I agree that the "updated" variable is wrongly named. It should probably be named up2date or something.)